core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusServiceSpec.js

```diff
@@ -0,0 +1,195 @@
+              needs_update: false
+            }}},
+        Second: {
+          content: {},
+          default_outcome: {},
+          feedback_1: {},
+          feedback_2: {}
+        },
+        Third: {
+          content: {
+            en: {
+              filename: 'filename3.mp3',
+              file_size_bytes: 100000,
+              needs_update: false
+            }}}};
+        return ContentIdsToAudioTranslationsObjectFactory.
+          createFromBackendDict(citatDict[stateName]);
+      },
+      getInteractionIdMemento: function(stateName) {
+        var interactionIds = {First: 'MultipleChoiceInput',
+          Second: 'MultipleChoiceInput',
+          Third: 'EndExploration'};
+        return interactionIds[stateName];
+      }
+    });
+    $provide.constant('INTERACTION_SPECS', {
+      MultipleChoiceInput: {
+        is_linear: false,
+        is_terminal: false
+      },
+      EndExploration: {
+        is_linear: false,
+        is_terminal: false
+      }
+    });
+  }));
+
+  describe('Translation status service', function() {
+    var tss = null;
+    beforeEach(inject(function($injector) {
+      tss = $injector.get('TranslationStatusService');
+      ess = $injector.get('ExplorationStatesService');
+      ContentIdsToAudioTranslationsObjectFactory = $injector.get(
+        'ContentIdsToAudioTranslationsObjectFactory');
+      StateContentIdsToAudioTranslationsService = $injector.get(
+        'StateContentIdsToAudioTranslationsService');
+
+      // To call _computeAllStatesStatus() function of
+      // TranslationStatusService, so that the status of all states is
+      // computed for the other dependent functions to work.
+      tss.getAllStateStatusColors();
+    }));
+
+    it('should return a correct list of state names for which ' +
+      'audio needs update', function() {
+      var statesNeedingAudioUpdate = tss.getAllStatesNeedUpdatewarning();
+      expect(Object.keys(statesNeedingAudioUpdate).length).toBe(1);
+      expect(statesNeedingAudioUpdate.First.
+        indexOf('Audio needs update!')).toBe(0);
+    });
+
+    it('should return a correct count of audio translations required ' +
+      'in an exploration', function() {
+      var explorationAudioRequiredCount = tss.
+        getExplorationAudioRequiredCount();
+      expect(explorationAudioRequiredCount).toBe(9);
+    });
+
+    it('should return a correct count of audio translations not available ' +
+      'in an exploration', function() {
+      var explorationAudioNotAvailableCount = tss.
+        getExplorationAudioNotAvailableCount();
+      expect(explorationAudioNotAvailableCount).toBe(6);
+    });
+
+    it('should correctly return an object contaning status colors of all ' +
+      'states in exploration', function() {
+      var stateWiseStatusColor = tss.getAllStateStatusColors();
+      expect(stateWiseStatusColor.First).toBe('#E9B330');
+      expect(stateWiseStatusColor.Second).toBe('#D14836');
```

Comment:
Can we define these string in a variable in beforeEach block? (Found it's used many times.)

Reply:
done